dex/candles/candles_test.go

```diff
@@ -167,41 +167,63 @@ func TestDelta(t *testing.T) {
 		EndRate:     175,
 	})
 
+	startCandle := &c.Candles[1]
+
+	testTime := tNow
 	check24 := func(expDelta float64, expVol uint64) {
+		t.Helper()
 		high, low := (expDelta + 0.01), (expDelta - 0.01)
-		delta24, vol24 := c.Delta(tNow.Add(-time.Hour * 24))
+		delta24, vol24 := c.Delta(testTime.Add(-time.Hour * 24))
 		if delta24 < low || delta24 > high {
-			t.Fatalf("wrong delta24. expected 0.75, got, %.3f", delta24)
+			t.Fatalf("wrong delta24. expected %.3f, got, %.3f", expDelta, delta24)
 		}
 		if vol24 != expVol {
-			t.Fatalf("wrong 24-hour volume. wanted 200, got %d", vol24)
+			t.Fatalf("wrong 24-hour volume. wanted %d, got %d", expVol, vol24)
 		}
 	}
 
 	// Basic function test.
 	check24(0.75, 200)
 
+	// Test halfway through the candle time. 125 -> 175 = +0.4, vol = 75 + 50
+	testTime = tNow.Add(time.Minute * 5 / 2)
+	check24(0.4, 125)
+
+	// Larger start rate tests undeflow handling.
+	startCandle.StartRate, startCandle.EndRate = startCandle.EndRate, startCandle.StartRate
+	check24(0.4, 125)
```

Comment:
Confirmed it fails without the patch with `wrong delta24. expected 0.400, got, -1.000`.
"underflow" spelled wrong, but np. :)